src/contracts/common/ExecutionBase.sol

```diff
@@ -21,7 +21,7 @@ contract Base {
     address public immutable source;
     bytes32 public immutable salt;
 
-    uint16 internal phasesWithDonations; //TODO remember to clear after call
+    uint16 internal lastPhaseDonated;
```

Comment:
Would really like to avoid a storage read / write in the ExecutionBase / ExecutionEnvironment contract.

Once TSTORE is live it'll be more palatable, but it'll take time for TSTORE to be rolled out on all the chains we're looking at for this.

Let's look for an alternative way to do it